### ISSUE TITLE
Caching e2e test are failing on ppc64le OCP

### ISSUE THREAD

**Issue by redmark-alt** [infinispan/infinispan-operator#778]
I'm running e2e testing in the 2.0.x branch on Openshift 4.5 ppc64le. The all the cache tests are failing but only warning or errors I can find in the pod logs are these.  Any idea how I can change the memory configuration.

Thanks

18:45:20,541 WARN  (blocking-thread--p3-t1) [org.infinispan.CONFIG] ISPN000586: The memory configuration element 'off-heap' has been deprecated. Please update your configuration
18:45:20,636 WARN  (blocking-thread--p3-t2) [org.infinispan.encoding.impl.StorageConfigurationManager] ISPN000599: Configuration for cache 'default' does not define the encoding for keys or values. If you use operations that require data conversion or queries, you should configure the cache with a specific MediaType for keys or values.

**Comment by dmvolod** [comment]
ISPN000586 and ISPN000599 are not problem related, they are usually printed in all cases
Could you please provide operator Pod logs

**Comment by redmark-alt** [comment]
Yes, it seems to indicate the cache was successful. The warning are the only differences I see between s390x and ppc64le and s390s is successful. 


[infinispan-operator-685c9989f6-8gpfq-infinispan-operator.log](https://github.com/infinispan/infinispan-operator/files/5889658/infinispan-operator-685c9989f6-8gpfq-infinispan-operator.log)

**Comment by dmvolod** [comment]
Seems to be operator don't have errors inside log. Please also upload log from running tests
What is the test error? You can also try to run a separate test providing it's name with env variable TEST_NAME=TestNodeStartup
Probably test timeout is not enough for passing all tests

**Comment by redmark-alt** [comment]
I'm running a single cache test now but I can't find a clear error other than connection timed out. I added some print lines into the test code.  Once odd thing I see the Readiness Probe has one IP address (10.130.2.213:11222) but the debug output has a different IP (192.168.88.4). Not sure if that is expected.

Readiness probe failed: Get https://10.130.2.213:11222/rest/v2/cache-managers/default/health/status: dial tcp 10.130.2.213:11222: connect: connection refused

Using project "datagrid-suite".
+ cd infinispan-operator
+ export EXPOSE_SERVICE_TYPE=LoadBalancer
+ EXPOSE_SERVICE_TYPE=LoadBalancer
+ export KUBECONFIG=/home/fedora/.kube/config
+ KUBECONFIG=/home/fedora/.kube/config
+ export RUN_LOCAL_OPERATOR=false
+ RUN_LOCAL_OPERATOR=false
+ export TESTING_NAMESPACE=datagrid-suite
+ TESTING_NAMESPACE=datagrid-suite
+ oc new-project datagrid-suite
Error from server (AlreadyExists): project.project.openshift.io "datagrid-suite" already exists
+ oc project datagrid-suite
Already on project "datagrid-suite" on server "https://api.test-ocp4-5.redhat.com:6443".
+ go mod vendor
+ go test -v ./test/e2e -run TestCacheService -timeout 60m
=== RUN   TestCacheService
Connection details test-cache-service-external developer ge2291SKXbOBSPB3 https
Error: Get "https://192.168.88.4:31981/rest/v2/cache-managers/default/health": dial tcp 192.168.88.4:31981: connect: connection timed out
Error: Get "https://192.168.88.4:31981/rest/v2/cache-managers/default/health": dial tcp 192.168.88.4:31981: connect: connection timed out
Error: Get "https://192.168.88.4:31981/rest/v2/cache-managers/default/health": dial tcp 192.168.88.4:31981: connect: connection timed out
--- FAIL: TestCacheService (468.89s)
panic: timed out waiting for the condition [recovered]
	panic: timed out waiting for the condition

[test-cache-service-0-infinispan.log](https://github.com/infinispan/infinispan-operator/files/5889752/test-cache-service-0-infinispan.log)

**Comment by dmvolod** [comment]
@redmark-alt are you running on OCP on AWS or have a LoadBalancer implementation on it?
Seems to be EXPOSE_SERVICE_TYPE not related to your OCP deployment. If you can access to all OCP nodes (both master and worker) you can try to use NodePort as EXPOSE_SERVICE_TYPE

**Comment by redmark-alt** [comment]
@dmvolod We are not testing ppc64le or s390x on AWS.  I'll give NodePort a try.  I don't believe s390x LoadBalancer either, we are using the exact same tests configuration(s).

I dropped in some more debug lines based in the comment and using host[0]. That why master-0 was used, my worker nodes don't start until "192.168.88.21"

// It should be ok to use the first node address available
hostAndPort = fmt.Sprintf("%s:%d", hosts[0], k.Kubernetes.GetNodePort(route))


hosts: [192.168.88.4 192.168.88.5 192.168.88.6 192.168.88.21 192.168.88.22 192.168.88.23 192.168.88.24 192.168.88.25 192.168.88.26 192.168.88.27 192.168.88.28 192.168.88.29 192.168.88.30]

**Comment by redmark-alt** [comment]
No luck with "EXPOSE_SERVICE_TYPE=NodePort" I'm getting the same failure.  All other e2e tests are passing its just the caching tests.

**Comment by dmvolod** [comment]
Not a problem with master or worker node as NodePode exposes can be used on any node. Something going wrong with NodePort remote server we've never tested. `NodePort` just using with TravisCI test.
It would be nice to support `EXPOSE_SERVICE_TYPE=Route` for this which is not yet implemented.

**Comment by redmark-alt** [comment]
@dmvolod Further investigation shows the problem is not ppc64le related. The cluster(s) we are using don't have nodes which are DNS serviced nor do they have FQDN(s).

**Comment by dmvolod** [comment]
Yes, @redmark-alt , correct. It's related to the OCP deployment. I was able to reprocure this problem OCP on AWS without LoadBalancer exposing. On some OCP deployments you can't access to the master and worker nodes directrly or by IP address and running tests with `EXPOSE_SERVICE_TYPE=Route` is a single working case. I submitted PR for this case. Hope it should help. You can fetch PR #782 from github and try to test it on your OCP instance

**Comment by redmark-alt** [comment]
I've run some simple tests and it looks good on singe cache test. I need to run the full e2e next. Big Thank you.

+ go test -v ./test/e2e -run TestCacheService -timeout 60m
=== RUN   TestCacheService
Wait for cache to be created: &{GET https://test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing/rest/v2/caches/default HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOnFVV2EzOTNDNTk5T0hDTEM=]] <nil> <nil> 0 [] false test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000124010}
Wait for cache to be created: &{GET https://test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing/rest/v2/caches/default HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOnFVV2EzOTNDNTk5T0hDTEM=]] <nil> <nil> 0 [] false test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000124010}
Wait for cache to be created: &{GET https://test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing/rest/v2/caches/default HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOnFVV2EzOTNDNTk5T0hDTEM=]] <nil> <nil> 0 [] false test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000124010}
Wait for cache to be created: &{GET https://test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing/rest/v2/caches/default HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOnFVV2EzOTNDNTk5T0hDTEM=]] <nil> <nil> 0 [] false test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000124010}
Wait for cache to be created: &{GET https://test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing/rest/v2/caches/default HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOnFVV2EzOTNDNTk5T0hDTEM=]] <nil> <nil> 0 [] false test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000124010}
Put request via route: &{POST https://test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing/rest/v2/caches/default/test HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOnFVV2EzOTNDNTk5T0hDTEM=] Content-Type:[text/plain]] {test-operator} 0x829f20 13 [] false test-cache-service-external-datagrid-suite.apps.test-ocp4-5.qe24.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000124010}
--- PASS: TestCacheService (107.15s)

**Comment by dmvolod** [comment]
The PR has merged into master and 2.0.x.
@redmark-alt please validate

**Comment by redmark-alt** [comment]
@dmvolod It took a few tries but finally the full e2e passed... I only tested the 2.0.x branch but I did it on OCP 4.5 & 4.6.

Question: As it is now when a single tests fails the test stops. Is there a option to keep moving to the next test?

Thank you,
-Ron

=== RUN   TestCacheService
Wait for cache to be created: &{GET https://test-cache-service-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/default HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOnVZQVBqUnh5ZVBTZHBMQGM=]] <nil> <nil> 0 [] false test-cache-service-external-datagrid-suite.apps.test-ocp4-6.qe20.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000040088}
Put request via route: &{POST https://test-cache-service-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/default/test HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOnVZQVBqUnh5ZVBTZHBMQGM=] Content-Type:[text/plain]] {test-operator} 0x829f20 13 [] false test-cache-service-external-datagrid-suite.apps.test-ocp4-6.qe20.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000040088}
--- PASS: TestCacheService (69.36s)
=== RUN   TestPermanentCache
Create cache: https://test-permanent-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
Put request via route: &{POST https://test-permanent-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test/test HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOkJoN2ZVUE1JV2xXbk95RjM=] Content-Type:[text/plain]] {test-operator} 0x829f20 13 [] false test-permanent-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000040088}
Delete cache: https://test-permanent-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
--- PASS: TestPermanentCache (179.93s)
=== RUN   TestCheckDataSurviveToShutdown
Create cache: https://test-data-shutdown-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
Put request via route: &{POST https://test-data-shutdown-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOlo5bmk1WUUxQ2N5TTZSWXE=] Content-Type:[application/xml;charset=UTF-8]] {<infinispan><cache-container><distributed-cache name ="test"><persistence><file-store/></persistence></distributed-cache></cache-container></infinispan>} 0x829f20 152 [] false test-data-shutdown-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000040088}
Put request via route: &{POST https://test-data-shutdown-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test/test HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOlo5bmk1WUUxQ2N5TTZSWXE=] Content-Type:[text/plain]] {test-operator} 0x829f20 13 [] false test-data-shutdown-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000040088}
Delete cache: https://test-data-shutdown-cache-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
--- PASS: TestCheckDataSurviveToShutdown (170.03s)
=== RUN   TestExternalService
Create cache: https://test-external-service-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
Put request via route: &{POST https://test-external-service-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test/test HTTP/1.1 1 1 map[Authorization:[Basic ZGV2ZWxvcGVyOlFHUUtjUlM2VU1xalZGRGU=] Content-Type:[text/plain]] {test-operator} 0x829f20 13 [] false test-external-service-external-datagrid-suite.apps.test-ocp4-6.qe20.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000040088}
Delete cache: https://test-external-service-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
--- PASS: TestExternalService (70.39s)
=== RUN   TestExternalServiceWithAuth
Create cache: https://text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
Create cache: https://text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
Put request via route: &{POST https://text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test/test HTTP/1.1 1 1 map[Authorization:[Basic Y29ubmVjdG9ydXNyOmNvbm5lY3RvcnBhc3M=] Content-Type:[text/plain]] {test-operator} 0x829f20 13 [] false text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000040088}
Delete cache: https://text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
Create cache: https://text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
Create cache: https://text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
Put request via route: &{POST https://text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test/test HTTP/1.1 1 1 map[Authorization:[Basic Y29ubmVjdG9ydXNyOmNvbm5lY3Rvcm5ld3Bhc3M=] Content-Type:[text/plain]] {test-operator} 0x829f20 13 [] false text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000040088}
Delete cache: https://text-external-service-with-auth-external-datagrid-suite.apps.test-ocp4-6.qe20.testing/rest/v2/caches/test
--- PASS: TestExternalServiceWithAuth (130.84s)
PASS
ok  	github.com/infinispan/infinispan-operator/test/e2e	1561.167s
